components/profile/unclaimed-hypercert-claim-button.tsx

```diff
@@ -26,42 +50,92 @@ export default function UnclaimedHypercertClaimButton({
       throw new Error("No wallet client found");
     }
 
-    if (!address) {
+    if (!account) {
       throw new Error("No address found");
     }
 
     if (
-      !allowListRecord.units ||
-      !allowListRecord.proof ||
-      !allowListRecord.token_id
+      !selectedHypercert?.units ||
+      !selectedHypercert?.proof ||
+      !selectedHypercert?.token_id
     ) {
       throw new Error("Invalid allow list record");
     }
-
-    // DUMMY VALUES
-    const root: `0x${string}` =
-      "0x0000000000000000000000000000000000000000000000000000000000000000";
+    await setDialogStep("preparing, active");
 
     console.log(allowListRecord);
 
-    const tx = await client.mintClaimFractionFromAllowlist(
-      BigInt(allowListRecord.token_id),
-      BigInt(allowListRecord.units),
-      allowListRecord.proof as `0x${string}`[],
-      undefined,
-    );
-    console.log(tx);
-    if (!tx) {
-      throw new Error("Failed to claim hypercert");
-    }
+    try {
+      await setDialogStep("claiming", "active");
+      const tx = await client.mintClaimFractionFromAllowlist(
+        BigInt(selectedHypercert?.token_id),
+        BigInt(selectedHypercert?.units),
+        selectedHypercert?.proof as `0x${string}`[],
+        undefined,
+      );
+      console.log(tx);
+
+      if (!tx) {
+        await setDialogStep("claiming", "error");
+        throw new Error("Failed to claim hypercert");
+      }
 
-    await waitForTransactionReceipt(walletClient, {
-      hash: tx,
-    });
+      await setDialogStep("confirming", "active");
+      const receipt = await waitForTransactionReceipt(walletClient, {
+        confirmations: 3,
+        hash: tx,
+      });
 
-    setTimeout(() => {
-      refresh();
-    }, 5000);
+      if (receipt.status == "success") {
+        await setDialogStep("route", "active");
+        const extraContent = createExtraContent(
+          receipt,
+          selectedHypercert?.hypercert_id!,
+          account.chain,
+        );
+        setExtraContent(extraContent);
+        await setDialogStep("done", "completed");
+        await revalidatePathServerAction([
+          `/hypercerts/${selectedHypercert?.hypercert_id}`,
```

Comment:
Same here about testing the revalidation :)

Reply:
Working on these in #420 

The issue appears to be in the API so taking that out of scope for this PR